src/Xamarin.Android.Build.Tasks/Tests/Xamarin.ProjectTools/Common/Builder.cs

```diff
@@ -220,19 +220,43 @@ protected bool BuildInternal (string projectOrSolution, string target, string []
 
 			LastBuildTime = DateTime.UtcNow - start;
 
-			LastBuildOutput = psi.FileName + " " + args.ToString () + Environment.NewLine;
+			var sb = new StringBuilder ();
+			sb.AppendLine ( psi.FileName + " " + args.ToString () + Environment.NewLine);
 			if (!ranToCompletion)
-				LastBuildOutput += "Build Timed Out!";
-			if (buildLogFullPath != null && File.Exists (buildLogFullPath))
-				LastBuildOutput += File.ReadAllText (buildLogFullPath);
-			LastBuildOutput += string.Format ("\n#stdout begin\n{0}\n#stdout end\n", p.StandardOutput.ReadToEnd ());
-			LastBuildOutput += string.Format ("\n#stderr begin\n{0}\n#stderr end\n", p.StandardError.ReadToEnd ());
-
-			var match = timeElapsedRegEx.Match (LastBuildOutput);
-			if (match.Success) {
-				LastBuildTime = TimeSpan.Parse (match.Groups ["TimeSpan"].Value);
+				sb.AppendLine ("Build Timed Out!");
+			if (buildLogFullPath != null && File.Exists (buildLogFullPath)) {
+				using (var fs = File.OpenRead (buildLogFullPath)) {
+					using (var sr = new StreamReader (fs, Encoding.UTF8, true, 65536)) {
+						string line;
+						while ((line = sr.ReadLine ()) != null) {
+							sb.AppendLine (line);
+						}
+					}
+				}
+			}
+			sb.AppendFormat ("\n#stdout begin\n{0}\n#stdout end\n", p.StandardOutput.ReadToEnd ());
+			sb.AppendFormat ("\n#stderr begin\n{0}\n#stderr end\n", p.StandardError.ReadToEnd ());
+
+			Console.WriteLine ("Searching for Time Elapsed");
+			using (var fs = File.OpenRead (buildLogFullPath)) {
```

Comment:
Can `buildLogFullPath` be null? It's checked against `null` both earlier and later in this method, but it's not checked *here*, and thus could plausibly result in an ArgumentNullException.

Should this `using` block be merged with the one above, on lines 228-235 of the new file? The earlier loop reads each line individually; we could simply merge the `Time Elapsed` check there, and use `line.StartsWith("Time Elapsed")` for good measure.